scripts/sct_utils.py

```diff
@@ -600,22 +600,22 @@ def create_folder(folder):
         return 1
 
 #=======================================================================================================================
-# check_if_3d
+# check_dim
 #=======================================================================================================================
 
-
-def check_if_3d(fname):
+def check_dim(fname, dim_lst=[3]):
     """
-    Check if input volume is 3d or less.
+    Check if input dimension matches the input dimension requirements specified in the dim list. 
+    Example: to check if an image is 2D or 3D: check_dim(my_file, dim_lst=[2, 3])
     :param fname:
     :return: True or False
     """
     from spinalcordtoolbox.image import Image
     dim = Image(fname).hdr['dim'][:4]
 
-    if not dim[0] <= 3:
-        printv('\nERROR: ' + fname + ' is not a 3D volume: {}. Exit program.\n'.format(','.join([str(d) for d in dim])), 
-                1, 'error')
+    if not dim[0] in dim_lst:
+        printv('\nERROR: ' + fname + ' does not match the input dimension requirements: {}. '
+                'Exit program.\n'.format(','.join([str(d) for d in dim])), 1, 'error')
```

Comment:
`format` can handle lists:
```python
>>> print("blabla {}".format([1, 2]))
blabla [1, 2]
```

Reply:
also, you might want to also display what is the requirements. E.g. something like this:
```python
printv("\nERROR: " + fname + " dimension (={}) does not match the dimension requirements (={}). Exit program.\n".format(dim, dim_lst), 1, 'error')
```

Reply:
Agree, thanks!

```
ERROR: /home/charley/2596/sub-issue2596_T1w.nii.gz dimension (=[  4 352 480 256]) does not match the input dimension requirements (=[2, 3]). Exit program.
```

Reply:
Ah that’s a bit ugly. I’d use:
~~~
printv("\nERROR: File " + fname + " has {} dimensions. Authorized dimensions are: {}. Exit program.\n".format(dim[0], dim_lst), 1, 'error')
~~~

Reply:
Agree!
```
ERROR: File /home/charley/2596/sub-issue2596_T1w.nii.gz has 4 dimensions. Authorized dimensions are: [2, 3]. Exit program.
```